src/Components/AllTrips.js

```diff
@@ -0,0 +1,162 @@
+import { useEffect, useState } from "react";
+import { onValue, ref, getDatabase } from "firebase/database";
+import AnimatedPage from "./AnimatedPage";
+import firebase from "../firebase";
+
+const AllTrips = ({
+  trips,
+  setTrips,
+  countryData,
+  searchValue,
+  setSearchValue,
+}) => {
+  const [filteredTrips, setFilteredTrips] = useState([]);
+
+  const handleSearchChange = (e) => {
+    if (
+      searchValue.citySearch &&
+      searchValue.countrySearch &&
+      e.target.id === "countrySearch"
+    ) {
+      setSearchValue({
+        citySearch: "",
+        countrySearch: e.target.value,
+      });
+    } else {
+      setSearchValue({
+        ...searchValue,
+        [e.target.id]: e.target.value,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (searchValue.citySearch && searchValue.countrySearch) {
+      const cityData = trips.filter((trip) => {
+        return trip.tripInfo.city.includes(searchValue.citySearch);
+      });
+
+      const newData = cityData.filter((trip) => {
```

Comment:
syntactically a bit odd to declare newData outside of the "if"? I think I see why you did it this way but I'm not sure if I've seen an if conditional broken up like this. just a thought!

Reply:
newData is still a part of the 'if'! I'll tweak my spacing a little bit to make all of the conditionals easier to read!! 🤩